Managers/UInputMan.cpp

```diff
@@ -1,2858 +1,1393 @@
+
+		// Detect and store mouse movement input, translated to analog stick emulation
+		int mousePlayer = MouseUsedByPlayer();
+		if (mousePlayer != PLAYER_NONE) {
+			m_AnalogMouseData.m_X += m_RawMouseMovement.m_X * 3;
+			m_AnalogMouseData.m_Y += m_RawMouseMovement.m_Y * 3;
+			// Cap the mouse input in a circle
+			m_AnalogMouseData.CapMagnitude(m_MouseTrapRadius);
+
+			// Only mess with the mouse pos if the original mouse position is not above the screen and may be grabbing the title bar of the game window
+			if (!m_DisableMouseMoving && !IsInMultiplayerMode()) {
+				if (m_TrapMousePos) {
+					// Trap the (invisible) mouse cursor in the middle of the screen, so it doesn't fly out in windowed mode and some other window gets clicked
+					position_mouse(g_FrameMan.GetResX() / 2, g_FrameMan.GetResY() / 2);
+				} else if (g_InActivity) {
+					// The mouse cursor is visible and can move about the screen/window, but it should still be contained within the mouse player's part of the window
+					ForceMouseWithinPlayerScreen(mousePlayer);
+				}
+			}
 
+			// Mouse wheel update, translate motion into discrete ticks
+			if (std::abs(mouse_z) >= 1) {
+				// Save the direction
+				m_MouseWheelChange = mouse_z;
+				// Reset the position
+				position_mouse_z(0);
+			} else {
+				m_MouseWheelChange = 0;
+			}
 
-void UInputMan::ClearAccumulatedStates()
-{
-	for (int element = 0; element < INPUT_COUNT; element++)
-	{
-		m_aNetworkAccumulatedElementPressed[element] = false;
-		m_aNetworkAccumulatedElementReleased[element] = false;
+			// Enable the mouse cursor positioning again after having been disabled. Only do this when the mouse is within the drawing area so it
+			// won't cause the whole window to move if the user clicks the title bar and unintentionally drags it due to programmatic positioning.
+			int mX = mouse_x / g_FrameMan.ResolutionMultiplier();
+			int mY = mouse_y / g_FrameMan.ResolutionMultiplier();
+			if (m_DisableMouseMoving && m_PrepareToEnableMouseMoving && (mX >= 0 && mX < g_FrameMan.GetResX() && mY >= 0 && mY < g_FrameMan.GetResY())) {
+				m_DisableMouseMoving = m_PrepareToEnableMouseMoving = false;
+			}
+		}
 	}
-}
 
-bool UInputMan::AccumulatedElementPressed(int element)
-{
-	if (element < 0 || element >= INPUT_COUNT)
-		return false;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	return m_aNetworkAccumulatedElementPressed[element];
-}
+	void UInputMan::UpdateJoystickInput() {
+		// Detect and store joystick button changed since last Update()
+		for (int joystick = 0; joystick < num_joysticks; ++joystick) {
+			for (int button = 0; button < joy[joystick].num_buttons; ++button) {
+				s_ChangedJoystickStates[joystick].button[button].b = joy[joystick].button[button].b != s_PrevJoystickStates[joystick].button[button].b;
+			}
 
+			// Determine deadzone settings
```

Comment:
I don't think we really need any of these deadzone comments, the code is fairly clear. This includes the one about disabling input if it's in the deadzone circle.

Also, maybe rename player to joystickPlayer or something so it's clear that it's the player who's using this joystick.